common/test/metadata/MetaDataMatcher.scala

```diff
@@ -40,4 +40,26 @@ object MetaDataMatcher extends Matchers  {
 
   }
 
+  def ensureDeepLink(result: Future[Result], expectedLink: String) {
+    val body = Jsoup.parseBodyFragment(contentAsString(result))
+    status(result) should be(200)
+    val script = body.getElementsByAttributeValue("href", expectedLink)
+    script.size() should be(1)
+  }
+
+  def ensureNoDeepLink(result: Future[Result], nonExistantLink: String) {
+    val body = Jsoup.parseBodyFragment(contentAsString(result))
+    status(result) should be(200)
+    val script = body.getElementsByAttributeValue("href", nonExistantLink)
+    script.size() should be(0)
```

Comment:
look like it's checking specifically for "ios-app://$appId/gnmguardian/$articleUrl?contenttype=front&source=google" and not just anything starting "ios-app://".

If we slightly changed the format of the url e.g. change the contenttype to ContentType then this test will always pass regardelss.

If we make a test to check no ios deep link it should really check that.

Having said that, maybe we don't consider it a big risk, but thought I'd let you know!